### DISCUSSION TITLE
v4.0.32 pre-release

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#48] [Announcements]
I've just released v4.0.32 as a pre-release. ~~Not being BJ I can't share this as an 'Announcement', rather just as a general discussion.~~ (BJ fixed it as an announcement, probably at 38,000 feet...)

Massive thanks to the following who caused it to happen, and who were most patient with testing (I hope I've not missed anyone):

@gcoan
@ProphetOfDoom
@ffm777
@Markyt77
@kernow

To give this pre-release a go, please make sure that 'Show beta versions' is on when re-downloading from HACS.

If you've made changes based on my suggestions since 4.0.31, then it may behoove you to back up the ```/homeassistant/custom_components/solcast_solar``` folder. I'm fairly certain I've not missed anything, so this should go smoothly. I just updated my 'prod' instance to 4.0.32 and all seems well so far.

Positive and negative feedback is most welcome.

(Happy travels, @BJReplay. We'll try and be good while you're away. Maybe.)

## What's Changed

- Bug fix: Independent API use counter for each Solcast account by @autoSteve
- Bug fix: Force all caches to /config/ for all platforms (fixes Docker deployments) #43 by @autoSteve
- Improve forecast fetch/retry logging debug, info, warning choice by @autoSteve
- Suppression of consecutive forecast fetches within fifteen minutes (fixes strange mutliple fetches should a restart occur exactly when automation for fetch is triggered) by @autoSteve
- Work-around: Prevent error when 'tally' is unavailable during retry by @autoSteve
- Fix for earlier HA versions not recognising version= for async_update_entry() #40 by @autoSteve

Full Changelog: https://github.com/BJReplay/ha-solcast-solar/compare/v4.0.31...v4.0.32

## Known issues

- The variable 'tally' should never be unavailable during a forecast fetch retry sequence, but it can be for some reason. This causes the site sensor to show as 'Unknown' until the retries are exhausted, or a successful fetch occurs. Other sensors are unaffected.

**Comment by gcoan** [comment]
> I've just released v4.0.32 as a pre-release. Not being BJ I can't share this as an 'Announcement', rather just as a general discussion.

Just installed it, restarted HA, all worked perfectly.  My scheduled update automation then ran 5 minutes later and all worked (200's, no retries).  I manually trigger the update again and it correctly suppressed the unnecessary update call.

I'll move my next scheduled update to 00:00 which should trigger some 429's, but all looking good so far, thanks @autoSteve

**Reply by autoSteve** [reply]
I moved _all_ my scheduled forecast gets to on the hour today, and **none** of them retried! Just when you want to force a failure, Solcast delights...

**Reply by BJReplay** [reply]
There's a major solar conference in Germany right now. The founder of Solcast is there - presumably spruiking their services. Maybe they turned up the hosting to eleven.

**Comment by BJReplay** [comment]
Announcement-ised from 11,500m approaching Broome.

**Reply by autoSteve** [reply]
Lols. I was 300 feet off...

**Reply by BJReplay** [reply]
I am IP banned from most of my self-hosted assets: apparently the in flight WiFi shares IP ranges with haxors. Oops. Can't even get in to unban myself.

**Reply by autoSteve** [reply]
VPN for the win...

**Reply by BJReplay** [reply]
Yeah, I think that there is some funky DNS hijacking going on. VPN is up, but can't reach unban control panel. Sleep looks like an option. Thanks for the great release.

**Comment by autoSteve** [comment]
Calling all testers to try out v4.0.32 and give it a thumbs up or down comment!

**Reply by gcoan** [reply]
Finally got a 429 from Solcast poll at 18:00 today.  Retries worked perfectly.

**Comment by ProphetOfDoom** [comment]
Just checked and the Tally values in the solcast.json file match the Solcast integration values for my NE and SW roofs.

All sensor values populated OK. No issues in HA log, so all looks good to me at least

Just to add, I upgraded HA to latest earlier today with no apparent issues
Core 2024.6.4
Supervisor 2024.06.0
Operating System 12.4
Frontend 20240610.1

**Comment by ffm777** [comment]
My sensors also updated just fine over the last two days, but I did get a tally warning this morning. See attached log file. But all sensors updated just fine.

[Solcast_Debug4.txt](https://github.com/user-attachments/files/15945151/Solcast_Debug4.txt)

**Reply by autoSteve** [reply]
Warning, and not error is great. That's the work-around doing its thing. Of why 'tally' goes missing I cant explain. @isorin may have an answer.

**Reply by isorin** [reply]
I've never look closely at 'tally' but I noticed it is only calculated when the data is updated and 'buildforecastdata' is called.
This means that after midnight and until the first pv_forecasts update tally won't exist.
I think we need to look into re-calculating it at midnight (the 'tally' should change at midnight to point to the new day).

**Reply by autoSteve** [reply]
What I and others have been seeing is it seems to go missing at the start of a forecast poll, then is okay. If a Solcast retry happens then any attempted access of it fails. I couldn't see why.

@ffm777's Solcast_Debug4.txt shows this happening, but not during a retry.

```
2024-06-23 08:53:59.186 INFO (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - Fetching forecast
2024-06-23 08:54:00.171 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - 'Tally' is currently unavailable for rooftop fbbf-8414-85fc-d840
2024-06-23 08:54:00.171 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - 'Tally' is currently unavailable for rooftop fbbf-8414-85fc-d840
2024-06-23 08:54:00.171 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - 'Tally' is currently unavailable for rooftop 8b6f-006d-1c6c-e224
2024-06-23 08:54:00.172 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - 'Tally' is currently unavailable for rooftop 8b6f-006d-1c6c-e224
2024-06-23 08:54:00.183 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - API returned data. API Counter incremented from 2 to 3
2024-06-23 08:54:00.183 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - writing usage cache
2024-06-23 08:54:00.186 INFO (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - Fetch successful
```

**Reply by gcoan** [reply]
What is Tally? Never heard of it, and don't see it mentioned in the readme

**Reply by autoSteve** [reply]
It's an internal variable that feeds the daily forecast sensor data (the sensor where LTS has vanished for you). When a new forecast comes in it gets updated, but what appears to be happening is that the variable goes missing at the start of a forecast gather, and stays missing until the gather is finished (whether that gather is successful or not).

**Reply by ProphetOfDoom** [reply]
It's the forecast kWh generation for the day for each roof id

![image](https://github.com/BJReplay/ha-solcast-solar/assets/109991286/d6758a84-70d7-4a9e-b440-d877f6893504)

![image](https://github.com/BJReplay/ha-solcast-solar/assets/109991286/54424813-7d78-4dad-937a-6ea808dbed95)

**Comment by autoSteve** [comment]
The silence is deafening with this pre-release. This is brilliant. Or nobody but a small handful is running it. 🤔

We do have a multi-API user, and other interested parties, through.

I'll give it a few more days before calling it, @BJReplay.

**Reply by Nilogax** [reply]
Thanks for the nudge @autoSteve - as mentioned in the dampening discussion, I've been on my tweaked version of this for about a week with no apparent issues.

**Reply by autoSteve** [reply]
🤭. Dampening bwoken reversion should be v4.0.34 at this stage from my POV. Seems good to me to back out that Oziee change based on both of our testing.

**Reply by BJReplay** [reply]
Seems to be working fine from here. ![Screenshot_20240625_131019_Home Assistant.jpg](https://github.com/BJReplay/ha-solcast-solar/assets/37993507/1c30e29a-8479-473f-9ef9-578f9a116ee0)

The graphs don't cope well with me being in a different time zone, but not admitting it to HA 🤣

**Reply by autoSteve** [reply]
That is MIGHTY night time solar generation, bloke. #jealous. Pity about peak power price bracket.

**Reply by ffm777** [reply]
v4.0.32 is working nicely on my end. Didn't even have a tally warning in the last two days.

**Reply by autoSteve** [reply]
Correction, @Nilogax, look for this reversion in the pre-release 4.0.33. Merged.
